# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/03 09:11:10+01:00 perex@suse.cz 
#   [ALSA] check __copy_to_user in sscape_upload_bootblock()
#   
#   Sound Scape driver
#   __copy_to_user() is called without checking its return value in
#   sound/isa/sscape.c::sscape_upload_bootblock .
#   
#   Signed-off-by: Jesper Juhl <juhl-lkml@dif.dk>
#   Signed-off-by: Takashi Iwai <tiwai@suse.de>
# 
# sound/isa/sscape.c
#   2004/11/30 07:37:33+01:00 perex@suse.cz +2 -1
#   [ALSA] check __copy_to_user in sscape_upload_bootblock()
#   
#   D:2004/11/30 14:37:33
#   C:Sound Scape driver
#   F:isa/sscape.c:1.17->1.18 
#   L:__copy_to_user() is called without checking its return value in
#   L:sound/isa/sscape.c::sscape_upload_bootblock .
#   Signed-off-by: Jesper Juhl <juhl-lkml@dif.dk>
#   Signed-off-by: Takashi Iwai <tiwai@suse.de>
# 
diff -Nru a/sound/isa/sscape.c b/sound/isa/sscape.c
--- a/sound/isa/sscape.c	2005-01-22 23:12:45 -08:00
+++ b/sound/isa/sscape.c	2005-01-22 23:12:45 -08:00
@@ -572,7 +572,8 @@
 			snd_printk(KERN_ERR "sscape: timeout reading firmware version\n");
 			ret = -EAGAIN;
 		} else {
-			__copy_to_user(&bb->version, &data, sizeof(bb->version));
+			if (__copy_to_user(&bb->version, &data, sizeof(bb->version)))
+				ret = -EFAULT;
 		}
 	}
 
